### DISCUSSION TITLE
Unable to export timeline data after recent timeline changes by google

### DISCUSSION THREAD

**Discussion by Pheggas** [Freika/dawarich#598] [General]
Hello.

As Google recently changed a way how timeline is saved (only on local device + encrypted backup on google servers), i am unable to export the timeline data by using Google Takeout. I only get 3 files in Location History folder: Encrypted Backups.txt, Settings.json and Tombstones.csv from which none has related data to timeline points or POIs.

the Encrypted Backups.txt has only simple text there: `You have encrypted Timeline backups stored on Google servers.` which i already know. There is basically no intended way to export timeline data from google, locally.

Do you have any methods that work?

**Comment by Wouter0100** [comment]
You are able to get an export through your phone.

Device settings > location > location services > timeline > export timeline data

But, I'm not sure yet how to import this data. Currently trying.

**Reply by Wouter0100** [reply]
A Google Phone Takeout import with that JSON file from your phone seems to work!

**Reply by Pheggas** [reply]
> Device settings > location > location services > timeline > export timeline data

Thank you for response. Unfortunately, i do not have the option to export timeline data there.

Edit: screenshot
![Screenshot_20241228-155929](https://github.com/user-attachments/assets/18d718e5-8293-4343-859f-ba411f516d2c)

**Reply by Pheggas** [reply]
I found it. Firstly i needed to update some google apps to latest version (Device settings -> Security & Privacy -> Check for updates -> proceed to update and reboot and then the options in location services changed and timeline become visible instead of some old location history sort of thing). I finally have done the export.

Anyway, is there a way to export semantic history?

**Reply by FoxxMD** [reply]
> A Google Phone Takeout import with that JSON file from your phone seems to work!

Can confirm this worked for me too.

**Reply by ItzSkyeYT** [reply]
> Device settings > location > location services > timeline > export timeline data

Thank you very much! 
For Samsung, this is how i did it: (i am unaware how it works for other phone manufacturers, but should be similar): 

1. Go to your phone's settings
2. Click on Location (on samsung it's in between 'security' and 'privacy and safety and emergency', or you could just do a search)
3. Then click on 'Location Services'. 
4. Normally there should be a 'Timeline' button in the list with the google icon.
5. Select your google account where your timeline is saved
6. Then you should be able to select 'Export Timeline data'. (Personally, ~14000 visits/routes saved was around 38MB)
8. You should be able to select where you want your file to be saved 
9. There you go! if you need to export it to another device you can always use Localsend/an external drive.
